pyked/converters.py

```diff
@@ -0,0 +1,741 @@
+        raise AttributeError('Error: at least one dataGroup needed.')
+
+    # all situations will have main experimental data in first dataGroup
+    dataGroup = dataGroups[0]
+    property_id = {}
+    unit_id = {}
+    # get properties of dataGroup
+    for prop in dataGroup.findall('property'):
+        unit_id[prop.attrib['id']] = prop.attrib['units']
+        property_id[prop.attrib['id']] = prop.attrib['name'].replace(' ', '-')
+        if property_id[prop.attrib['id']] not in [
+            'temperature', 'pressure', 'ignition-delay', 'pressure-rise'
+            ]:
+            raise KeyError(property_id[prop.attrib['id']] + ' not valid dataPoint property')
+    if property_id == {}:
+        raise AttributeError('Error: at least one property needed in dataGroup.')
+
+    # now get data points
+    datapoints = []
+    for dp in dataGroup.findall('dataPoint'):
+        datapoint = {}
+        for val in dp:
+            units = unit_id[val.tag]
+            if units == 'Torr':
+                units = 'torr'
+            datapoint[property_id[val.tag]] = [val.text + ' ' + units]
+        datapoints.append(datapoint)
+
+    if len(datapoints) == 0:
+        raise AttributeError('Error: need at least one dataPoint.')
+
+    # RCM files may have a second dataGroup with volume-time history
+    if len(dataGroups) == 2:
+        assert root.find('apparatus/kind').text == 'rapid compression machine', \
+               'Second dataGroup only valid for RCM.'
+
+        assert len(datapoints) == 1, 'Multiple datapoints for single volume history.'
+
+        dataGroup = dataGroups[1]
+        for prop in dataGroup.findall('property'):
+            if prop.attrib['name'] == 'time':
+                time_dict = {'units': prop.attrib['units'], 'column': 0}
+                time_tag = prop.attrib['id']
+            elif prop.attrib['name'] == 'volume':
+                volume_dict = {'units': prop.attrib['units'], 'column': 1}
+                volume_tag = prop.attrib['id']
+
+        volume_history = {'time': time_dict, 'volume': volume_dict, 'values': []}
+
+        # collect volume-time history
+        for dp in dataGroup.findall('dataPoint'):
+            time = None
+            volume = None
+            for val in dp:
+                if val.tag == time_tag:
+                    time = float(val.text)
+                elif val.tag == volume_tag:
+                    volume = float(val.text)
+            volume_history['values'].append([time, volume])
+
+        datapoints[0]['volume-history'] = volume_history
+
+    elif len(dataGroups) > 2:
+        raise NotImplementedError('More than two DataGroups not supported.')
+
+    return datapoints
+
+
+def convert_ReSpecTh(filename_xml, output='', file_author='', file_author_orcid=''):
+    """Convert ReSpecTh XML file to ChemKED YAML file.
+
+    Args:
+        filename_xml (`str`): Name of ReSpecTh XML file to be converted.
+        output (`str`, optional): Output path for converted file.
```

Comment:
If it is going to be joined with the `filename_ck`, then I think it should be `output_folder`. But I don't know that we need this argument at all, since I think if people want a folder then can include it in `filename_ck` (but maybe need to check that). Is there a reason you want a folder argument like this?